src/main/java/core/basesyntax/service/impl/FruitTransactionMapper.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.enums.Operation;
+import core.basesyntax.service.Mapper;
+
+public class FruitTransactionMapper implements Mapper<FruitTransaction, String> {
+    private static final String DELIMITER = ",";
+    private static int OPERATION_INDEX = 0;
+    private static int FRUIT_INDEX = 1;
+    private static int QUANTITY_INDEX = 2;
+
+    @Override
+    public FruitTransaction toObject(String string) {
+        String[] split = string.split(DELIMITER);
+        validateSplitLength(split, string);
+        Operation operation = Operation.fromCode(split[OPERATION_INDEX]);
```

Comment:
Ensure that the `Operation.fromCode` method properly handles invalid operation codes. If it doesn't, you should include a check here to ensure that the `operation` is not `null` after this line, throwing an exception if it is.